spring-boot/proxy/src/main/java/eu/coatrack/proxy/filters/pre/ApiKeyAuthFilter.java

```diff
@@ -54,14 +55,31 @@
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
         log.debug(String.format("Api key auth filter received request of type %s", servletRequest.getClass()));
 
+        Authentication authToken = null;
         if (servletRequest instanceof HttpServletRequest) {
             final HttpServletRequest req = (HttpServletRequest) servletRequest;
```

Comment:
Here there could be a submethod called `retrieveKeyParameterFromRequest` or `getKeyParameterFromRequest.
As from what I understood, the keyParam is just being taken from the request and then a authToken is generated in order to do the authentication